models/Node.js

```diff
@@ -43,8 +43,8 @@ function Node(name, type, objectId, frameId, nodeId) {
         this.publicData = {};
     } else {
         let nodeTemplate = nodeTypes[type];
-        this.privateData = nodeTemplate.properties.privateData || {};
-        this.publicData = nodeTemplate.properties.publicData || {};
+        this.privateData = JSON.parse(JSON.stringify(nodeTemplate.properties.privateData)) || {};
```

Comment:
The || should be inside the calls to stringify since JSON.parse(JSON.stringify(undefined)) is a syntax error